node/src/components/rpc_server/http_server.rs

```diff
@@ -56,32 +54,13 @@ pub(super) async fn run<REv: ReactorEventT>(
     QueryBalance::register_as_handler(effect_builder, api_version, &mut handlers);
     let handlers = handlers.build();
 
-    let make_svc = hyper::service::make_service_fn(move |_| {
-        let service_routes = casper_json_rpc::route(RPC_API_PATH, max_body_bytes, handlers.clone());
-
-        // Supports content negotiation for gzip responses. This is an interim fix until
-        // https://github.com/seanmonstar/warp/pull/513 moves forward.
-        let service_routes_gzip = warp::header::exact(ACCEPT_ENCODING.as_str(), "gzip")
-            .and(service_routes.clone())
-            .with(warp::compression::gzip());
-
-        let service = warp::service(service_routes_gzip.or(service_routes));
-        async move { Ok::<_, Infallible>(service.clone()) }
-    });
-
-    let make_svc = ServiceBuilder::new()
-        .rate_limit(qps_limit, Duration::from_secs(1))
-        .service(make_svc);
-
-    let server = builder.serve(make_svc);
-    info!(address = %server.local_addr(), "started JSON-RPC server");
-
-    let (shutdown_sender, shutdown_receiver) = oneshot::channel::<()>();
-    let server_with_shutdown = server.with_graceful_shutdown(async {
-        shutdown_receiver.await.ok();
-    });
-
-    let _ = tokio::spawn(server_with_shutdown).await;
-    let _ = shutdown_sender.send(());
-    trace!("JSON-RPC server stopped");
+    super::rpcs::run(
```

Comment:
Out of curiosity, why do we want to extract this logic to a new method vs leave it inline here?

Reply:
never mind, i see it getting reused below.